apim-operator/pkg/controller/api/api_controller.go

```diff
@@ -321,22 +318,41 @@ func (r *ReconcileAPI) Reconcile(request reconcile.Request) (reconcile.Result, e
 		}
 	}
 
-	// gets the data from analytics secret
-	analyticsData, err := getSecretData(r)
-
-	if err == nil && analyticsData != nil && analyticsData[usernameConst] != nil &&
-		analyticsData[passwordConst] != nil && analyticsData[certConst] != nil {
-		analyticsEnabled = "true"
-		analyticsUsername = string(analyticsData[usernameConst])
-		analyticsPassword = string(analyticsData[passwordConst])
-		analyticsCertSecretName := string(analyticsData[certConst])
-
-		log.Info("Finding analytics cert secret " + analyticsCertSecretName)
-		// Check if this secret exists and append it to volumes
-		jobVolumeMountTemp, jobVolumeTemp, errCert := analyticsVolumeHandler(analyticsCertSecretName, r, jobVolumeMount, jobVolume)
-		if errCert == nil {
-			jobVolumeMount = jobVolumeMountTemp
-			jobVolume = jobVolumeTemp
+	// gets analytics configuration
+	analyticsConf, analyticsEr := getConfigmap(r, analyticsConfName, wso2NameSpaceConst)
+	if analyticsEr != nil {
+		log.Info("Disabling analytics since analytics configurations were not found")
```

Comment:
Shall we improve this?

Disabling analytics since the analytics configuration related config map not found.

Reply:
Changed